Iterable-iOS-SDK/IterableAPI.m

```diff
@@ -346,6 +346,52 @@ - (instancetype)createSession:(NSDictionary *)launchOptions useCustomLaunchOptio
 //////////////////////////////////////////////////////////////
 
 // documented in IterableAPI.h
++ (IterableAPI *)sharedInstance
+{
+    if (sharedInstance == nil) {
+        LogError(@"[sharedInstance called before sharedInstanceWithApiKey");
+    }
+    return sharedInstance;
+}
+
+// documented in IterableAPI.h
++ (IterableAPI *)sharedInstanceWithApiKey:(NSString *)apiKey andEmail:(NSString *)email launchOptions:(NSDictionary *)launchOptions
+{
+    // threadsafe way to create a static singleton https://stackoverflow.com/questions/5720029/create-singleton-using-gcds-dispatch-once-in-objective-c
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[IterableAPI alloc] initWithApiKey:apiKey andEmail:email launchOptions:launchOptions];
+    });
+    return sharedInstance;
+}
+
+// documented in IterableAPI.h
++ (IterableAPI *)sharedInstanceWithApiKey:(NSString *)apiKey andUserId:(NSString *)userId launchOptions:(NSDictionary *)launchOptions
+{
+    // threadsafe way to create a static singleton https://stackoverflow.com/questions/5720029/create-singleton-using-gcds-dispatch-once-in-objective-c
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[IterableAPI alloc] initWithApiKey:apiKey andUserId:userId launchOptions:launchOptions];
+    });
+    return sharedInstance;
+}
+
+// documented in IterableAPI.h
++(void) getAndTrackDeeplink:(NSURL *)webpageURL callbackBlock:(ITEActionBlock)callbackBlock
+{
+    if ([webpageURL.absoluteString rangeOfString:ITBL_DEEPLINK_IDENTIFIER].location != NSNotFound) {
```

Comment:
Would it be difficult to do a regex to have it search for `/a/UUID` (i.e. `links.munchery.com/a/e013a3e6866b4142b6807c3832ec16d5` perhaps this regex `\/a\/[a-zA-Z0-9]+` ).
Just an extra safeguard against anyone having `/a/` anywhere in their url.